### ISSUE TITLE
Shoud the router run after instanciation ?

### ISSUE THREAD

**Issue by hyptos** [ramiel/router.js#20]
Hi,

I had the problem where I was wondering when I land on a page with a routed url like : 

www.foo.com/#/bar

```
var router = new Router()
	.addRoute('/:bar', function(req, next){
		var bar = req.params.bar;
                console.log(bar);
 });
```

Why wasn't I routed to /bar and see the log ?

I found out in the documentation that after instanciation, I need to add the .run(); in order to catch this right away.

```
var router = new Router()
	.addRoute('/:bar', function(req, next){
		var bar = req.params.bar;
                console.log(bar);
 }).run();
```

So in order to make it more user(dev) "friendly" shouldn't it run instead of having the dev to explicitly do it ?

I think the other way around is less likely to happen no ? If you want to not catch a route then you would need to add the pause after instanciation.

What do you think ? Or why did you make it this way ? Is there any reason ?

**Comment by ramiel** [comment]
Hi,
first of all, apart of the `run` problem, do you managed to have the route called? Because I expect to read:

```javascript
router.addRoute('#/:bar', function(req, next){
...
}
```

with the `#` before the slash.
By the way, if you have problem having this route working let me know.  
Talking about the `run` function. I decided to not have the router automatically called because I had some cases where the application should not start from the current url. Actually these application, whatever was the url, should start from `#/`, so, after the definition of the routes often I had `router.run('#/')`.
So basically the reason is "I had some use cases".
Then I prefer to have it explicitly written in the application through `run`. If you want to use `pause` instead then you have to remember to use `play` to restart the router or your application won't catch the url changing anymore.This solution looks tricky to me.
And as last, `run` let you define which url to start from (passing a parameter) which cover all the use cases definitely.
Let me know if you find this arguments poor so we could talk deeply.

**Comment by hyptos** [comment]
I don't remember the specifics (not at work) but yeah the router is working nicely just the run thingy.

It was just a question I had, in order to make things easier but it's mainly because I fired `run()` without any argument, since we can do it like you said with the `router.run('#/')`. I think my point is moot :)

Before closing this, could you put a link or something to the [IE8 branch ](https://github.com/ramiel/router.js/tree/feature/ie8) I spent like 1 hour trying to make the release branch work with a polyfill for pushstate until I found an older issue where you said that there is an IE8 branch ><

Thanks for your time.

Bonne soirée  (je n'avais pas vu que tu étais Français :p)

**Comment by ramiel** [comment]
Thanks for the point, I will eventually add the link! I'm happy you have an answer for your question!

Bonne soirée a toi aussi, je suis pas français mais je vive en france!
